backend/algorithms/recommendations.py

```diff
@@ -11,14 +11,38 @@
 
 NOTE: For sample user data, look at exampleUsers.json to get an idea of the structure
 """
-
+from .conditions import User, create_condition, CACHED_CONDITIONS_TOKENS
+import json
 def get_unlocked_courses(user):
     """
     Given a user's data, returns a list of all the unlocked courses which are
     ACCURATE. Refer to the getAllUnlocked() API for an example of how this works. 
 
     OUTPUT: [COMP1511, COMP1521, COMP1531...]
     """
+
+    COURSE_PATH = "./data/finalData/conditionsTokens.json"
+    PICKLE_FILE = "./algorithms/conditions.pkl"
+    course_list = {}
+    result = []
+    with open(COURSE_PATH, encoding='UTF-8') as f:
+        course_list = json.load(f)
+        f.close()
+    
+    # Load all the necessary conditions
+    cached_conditions = {} # Mapping course to condition object
```

Comment:
Best to have cached_conditions initialised outside the function so it's only created once since the Condition objects themselves don't change